### PR TITLE
Make pep561 tests parrallel and sandboxed

### PR DESCRIPTION
**Pull request by ethanhs**
Each test is run in a seperate virtualenv which is deleted after the
tests. This means they can be run in parrallel, and caused about a 2x
speedup for pep561 on my machine (4x parrallel). This also will allow for the tests
to scale better with egg support, and bypass issues with uninstalling
some types of packages.

### CODE REVIEW

**Comment by ethanhs** [comment]
This is ready for review now. It isn't *as* important as I decided egg support serves little purpose, but it provides a nice speedup, and makes things cleaner.

**Comment by ethanhs** [comment]
I just realized I might have been a bit vague in my last message. I'm hoping to get this merged soonish so that I can work on adding support for editable installs of packages (which I already have an implementation of), then support for partial stub packages. If someone finds the time I'd appreciate a review.

File: mypy/test/testpep561.py [hunk 1]
```diff
@@ -19,112 +22,112 @@
...
         if returncode != 0:
             self.fail('\n'.join(lines))
-        try:
-            yield
-        finally:
-            returncode, lines = run_command([python_executable, '-m', 'pip', 'uninstall',
-                                             '-y', pkg], cwd=package_path)
-            if returncode != 0:
-                self.fail('\n'.join(lines))
+        yield
```
Comment (ilevkivskyi) [inline mypy/test/testpep561.py]: I am not sure what is the benefit of context manager which has yield as a last statement. Can't be this just made a function?

Comment (ethanhs) [reply mypy/test/testpep561.py]: Good point. I will refactor things into just using this as a function.

**Pull request merged**